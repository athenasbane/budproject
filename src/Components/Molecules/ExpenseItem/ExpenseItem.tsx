import React from 'react';
import { ITransaction } from '../../../types/Interfaces/Api.interface';
import { currencySymbol } from '../../../lib/currencySymbol.lib';
import { Container, Title, ExpenseContainer, Circle } from '../../Atoms';

const ExpenseItem = ({ amount, description, category_title }: ITransaction) => {
    return (
        <ExpenseContainer>
            <Container direction="row">
                <Circle>{description.at(0)}</Circle>
                <Container marginLeft={15} direction="column">
                    <Title backgroundColor="white" textAlign="left" textSize="medium">
                        {description}
                    </Title>
                    <Title backgroundColor="white" textAlign="left" textSize="small">
                        {category_title}
                    </Title>
                </Container>
            </Container>
            <Container>
                <Title backgroundColor="white" textSize="small" textAlign="right">
                    {currencySymbol[amount.currency_iso]}
                    {Math.abs(amount.value)}
                </Title>
            </Container>
        </ExpenseContainer>
    );
};

export default ExpenseItem;
