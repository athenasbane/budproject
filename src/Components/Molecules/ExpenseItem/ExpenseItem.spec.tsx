import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import ExpenseItem from './ExpenseItem';
import { mockExpense } from '../../../__mocks__/expense.mock';

describe('<ExpenseItem />', () => {
    it('should match snapshot', () => {
        const tree = renderer.create(<ExpenseItem {...mockExpense} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
