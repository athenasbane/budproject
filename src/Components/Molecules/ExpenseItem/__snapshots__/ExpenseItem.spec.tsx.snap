// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpenseItem /> should match snapshot 1`] = `
<article
  className=""
>
  <div
    className=""
    direction="row"
  >
    <h1
      className=""
    >
      T
    </h1>
    <div
      className=""
      direction="column"
    >
      <h1
        className=""
      >
        Tesco
      </h1>
      <h1
        className=""
      >
        Groceries
      </h1>
    </div>
  </div>
  <div
    className=""
  >
    <h1
      className=""
    >
      Â£
      57.21
    </h1>
  </div>
</article>
`;
